src/circuit/gadgets/constraints.rs

```diff
@@ -166,7 +166,7 @@ pub fn div<F: PrimeField, CS: ConstraintSystem<F>>(
             inv.map(|i| tmp.mul_assign(i));
             Ok(tmp)
         } else {
-            Err(SynthesisError::DivisionByZero)
+            panic!("Division by zero.");
```

Comment:
I'm not sure about this change. The error is the right outcome, but if this change was correct it should never happen, right? If that is your intention with the explicit panic, then maybe this should be `unreachable!()`.

Reply:
I wanted to guarantee it was not happening. I reverted this modification.